src/features/goal/utils/date.ts

```diff
@@ -0,0 +1,8 @@
+export const formatDate = (splitedDate: string[], separator: string) => {
+  return splitedDate.join(separator);
+};
+
+export const isValidDate = (year: string, month: string, day: string) => {
+  const date = year + '-' + month + '-' + day;
+  return !isNaN(Date.parse(date));
+};
```

Comment:
쥐피티 형에 따르면 이 함수로는 validate가 안되는 케이스가 있다네요~~! 예: 2월 31일, leap years 등

AI 추천 코드 전달드립니다~🎊

``` typescript
export const isValidDate = (year: string, month: string, day: string) => {
  const yearNum = parseInt(year, 10);
  const monthNum = parseInt(month, 10) - 1; // Subtract 1 because months are 0-indexed
  const dayNum = parseInt(day, 10);

  const parsedDate = new Date(yearNum, monthNum, dayNum);

  return (
    parsedDate.getFullYear() === yearNum &&
    parsedDate.getMonth() === monthNum &&
    parsedDate.getDate() === dayNum
  );
};
```

Reply:
헉 이렇게 하다가 변경한건데 다시 돌려놓겠슴다~! 감사감사 🙏